front/components/sections/Calculator/helper.ts

```diff
@@ -1,35 +1,44 @@
+import { strapiAppointmentQuery } from 'lib/strapi/appointments/queryType';
 import { formData } from '.';
-import { strapiTherapistQuery } from 'lib/strapi/therapists/queryType';
+import { strapiTherapistsQuery } from 'lib/strapi/therapists/queryType';
 
-export function calculation({ data, therapistList, cost, discount }: props) {
+export function calculation({ data, therapistList, appointmentList, cost, discount }: props) {
     discount = 0;
     cost = 0;
+    let numberOfDataSession = Number(data?.session);
     if (data?.workshop) {
-        data.session = Number(data.session) - 1;
+        numberOfDataSession = numberOfDataSession - 1;
     }
-    for (let i = 0; i < Number(therapistList?.length); i++) {
-        if (therapistList?.[i].attributes.first_name === data?.therapist) {
-            cost = Number(therapistList?.[i].attributes.session_cost) * Number(data?.session);
+    therapistList?.find((therapist) => {
+        if (therapist?.id === Number(data?.therapist)) {
```

Comment:
```suggestion
        if (therapist?.id !== Number(data?.therapist)) return;
```
This warden will save you from one nesting